### PR TITLE
Refactor Point, add hashCode, a few fixes.

### PR DESCRIPTION
**Pull request by stolstov**
For issue https://github.com/Esri/geometry-api-java/issues/236

I wanted to change internals of the Point class, and this was a good chance.
Some fixes to ensure the objects that are equal have same hashCode value (miscompare happens for  NaN values in the attributes).

### CODE REVIEW

**Comment by stolstov** [comment]
Small change in behavior - getter methods on Point class ordinates (getX, etc) would not throw exception if point is empty.

**Comment by stolstov** [comment]
@alocke Could you review?

File: src/main/java/com/esri/core/geometry/Envelope.java [hunk 1]
```diff
@@ -1030,7 +1010,7 @@ public boolean equals(Object _other) {
 	public int hashCode() {
 		int hashCode = m_description.hashCode();
 		hashCode = NumberUtils.hash(hashCode, m_envelope.hashCode());
-		if (!isEmpty() && m_attributes != null) {
+		if (!isEmpty()) {
```
Comment (alocke) [inline src/main/java/com/esri/core/geometry/Envelope.java]: Are we sure at this point that m_attributes != null?

Comment (stolstov) [reply src/main/java/com/esri/core/geometry/Envelope.java]: There should not be a problem here. I am making sure that m_attributtes are not null if the geometry has any extra attributes (z, m, etc).

**Pull request merged**